src/components/widget/ShortenField.jsx

```diff
@@ -0,0 +1,38 @@
+import style from './ShortenField.module.css';
+import { Button } from '../elements/Button';
+import { useState, useRef } from 'react';
+
+function ShortenField() {
+  const [showError, setShowError] = useState(false);
+  const inputRef = useRef(null);
+
+  const handleShort = () => {
+    const urlPattern =
+      /[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_+.~#?&//=]*)/gi;
```

Comment:
OMG :fearful:

Move this scary peace of code outside of component please. You don't need to create it each function call.